File: PRACTICA1/JAVA/deepspace/src/deepspace/EnemyStarShip.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author Lucía Salamanca López
 *         Celia Arias Martínez
 */
public class EnemyStarShip implements SpaceFighter{
    private float ammoPower;
    private String name;
    private float shieldPower;
    private Damage damage;
    private Loot loot;
    

    EnemyStarShip(String n, float a, float s, Loot l, Damage d){
        name = n; 
        ammoPower = a;
        shieldPower = s;
        loot = new Loot(l.getNSupplies(), l.getNWeapons(), l.getNShields(), l.getNHangars(), l.getNMedals(), l.getEfficient(), l.spaceCity()); 
        damage = d; 
    }

    EnemyStarShip(EnemyStarShip e){
       name = e.getName();
       ammoPower = e.getAmmoPower();
       shieldPower = e.getShieldPower();
       loot = e.getLoot();
       damage = e.getDamage();
    }
    EnemyToUI getUIversion(){
       return new EnemyToUI(this);
    }
    
    @Override
    public float fire(){
       return getAmmoPower(); 
    }
    public float getAmmoPower(){
        return ammoPower;
    }
    public Damage getDamage(){
        return damage;
    }
    public Loot getLoot(){
        return loot;
    }
    public String getName(){
        return name;
    }
    public float getShieldPower(){
        return shieldPower;
    }
    @Override
    public float protection(){
        return getShieldPower();
    }
    @Override
    public ShotResult receiveShot(float shot){
        if (protection() < shot)
            return ShotResult.DONOTRESIST;
        return ShotResult.RESIST;
    }
    public String toString(){
        String mensaje = "Name: "+name+ ", Ammo Power: "+ammoPower+", Shield Power: "+shieldPower+"\n\tLoot..:"+loot.toString()+"\n\tDamage: "+damage.toString();
        return mensaje;
    }
}

File: PRACTICA1/RUBY/Deepspace/EnemyStarShip.rb
#encoding:utf-8

module Deepspace


# @author Lucía Salamanca López, Celia Arias Martínez
class EnemyStarShip

def initialize(n,a,s,l,d)
  @ammoPower = a
  @name = n
  @shieldPower = s
  @loot = l
  @damage = d
end

def self.newCopy(e)
  new(e.name, e.ammoPower, e.shieldPower, e.loot, e.damage)
end

def getUIversion
  EnemyToUI
end

def fire
  ammoPower
end

def ammoPower
  @ammoPower
end

def damage
  @damage
end

def loot
  @loot
end

def name
  @name
end

def shieldPower
  @shieldPower
end

def protection
  shieldPower
end

def receiveShot(shot)
  if (protection < shot)
    return ShotResult::DONOTRESIST
  else
    return ShotResult::RESIST
  end
end

def to_s
  return "Name: #{@name}, " + \
      "Ammo Power: #{@ammoPower}, " + \
      "Shield Power: #{@shieldPower}\n" + \
      "\tLoot..: #{@loot}\n" + \
      "\tDamage: #{@damage}"
end

end

end
